.github/workflows/push_pr.yaml

```
@@ -33,35 +33,35 @@ jobs:
           echo "${target_head} = ${merge_base}"
           exit 0
 
-  test:
+  go-test:
     name: Run tests
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
-      - uses: actions/setup-go@v4
+      - uses: actions/setup-go@v5
         with:
-          go-version: '1.19'
-      - uses: actions/cache@v4
-        with:
-          path: |
-            ~/.cache/go-build
-            ~/go/pkg/mod
-          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
-          restore-keys: |
-            ${{ runner.os }}-go-
+          go-version-file: "go.mod"
       - name: Run tests
         # CLI autodetects whether it is running on GHA or not. If we let GHA set this to true, as it normally does,
         # tests cases where GITHUB_ACTIONS is expected to be unset would fail.
         run: GITHUB_ACTIONS=false go test -race ./...
 
+  check-changelog:
+    name: Run tests
```

Copy&Paste leftover?
```suggestion
    name: Check changelog
```